tests/testthat/test-expect_type_linter.R

```diff
@@ -1,48 +1,52 @@
 test_that("expect_type_linter skips allowed usages", {
+  linter <- expect_type_linter()
+
   # expect_type doesn't have an inverted version
-  expect_lint("expect_true(!is.numeric(x))", NULL, expect_type_linter())
+  expect_lint("expect_true(!is.numeric(x))", NULL, linter)
   # NB: also applies to tinytest, but it's sufficient to test testthat
-  expect_lint("testthat::expect_true(!is.numeric(x))", NULL, expect_type_linter())
+  expect_lint("testthat::expect_true(!is.numeric(x))", NULL, linter)
 
   # other is.<x> calls are not suitable for expect_type in particular
-  expect_lint("expect_true(is.data.frame(x))", NULL, expect_type_linter())
+  expect_lint("expect_true(is.data.frame(x))", NULL, linter)
 
   # expect_type(x, ...) cannot be cleanly used here:
-  expect_lint("expect_true(typeof(x) %in% c('builtin', 'closure'))", NULL, expect_type_linter())
+  expect_lint("expect_true(typeof(x) %in% c('builtin', 'closure'))", NULL, linter)
 
   # expect_type() doesn't have info= or label= arguments
-  expect_lint("expect_equal(typeof(x), t, info = 'x should have type t')", NULL, expect_type_linter())
-  expect_lint("expect_equal(typeof(x), t, label = 'x type')", NULL, expect_type_linter())
-  expect_lint("expect_equal(typeof(x), t, expected.label = 'type')", NULL, expect_type_linter())
-  expect_lint("expect_true(is.double(x), info = 'x should be double')", NULL, expect_type_linter())
+  expect_lint("expect_equal(typeof(x), t, info = 'x should have type t')", NULL, linter)
+  expect_lint("expect_equal(typeof(x), t, label = 'x type')", NULL, linter)
+  expect_lint("expect_equal(typeof(x), t, expected.label = 'type')", NULL, linter)
+  expect_lint("expect_true(is.double(x), info = 'x should be double')", NULL, linter)
 })
 
 test_that("expect_type_linter blocks simple disallowed usages", {
+  linter <- expect_type_linter()
+
   expect_lint(
     "expect_equal(typeof(x), 'double')",
     rex::rex("expect_type(x, t) is better than expect_equal(typeof(x), t)"),
```

Comment:
this is repeated too, right?

Reply:
Yes, but there is also `rex::rex("expect_type(x, t) is better than expect_true(is.<t>(x))")` in the same unit, so I didn't create a new variable for lint message.

Reply:
OK, we can skip for now